From ca91f57e637b36cc2cad8b8def53997c7fabdb08 Mon Sep 17 00:00:00 2001
From: "Darryl L. Miles" <darryl.miles@darrylmiles.org>
Date: Tue, 15 Jul 2025 23:54:55 +0100
Subject: [PATCH 1/2] F42: fix for etext symbol glibc 2.41+

The man page end(3) changed as well over time to support the new definition.
---
 utils/netlist.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/utils/netlist.c b/utils/netlist.c
index b835732f..b1556c85 100644
--- a/utils/netlist.c
+++ b/utils/netlist.c
@@ -22,6 +22,7 @@
 static char rcsid[] __attribute__ ((unused)) = "$Header: /usr/cvsroot/magic-8.0/utils/netlist.c,v 1.1.1.1 2008/02/03 20:43:50 tim Exp $";
 #endif	/* lint */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include "utils/magic.h"
@@ -369,6 +370,8 @@ NLNetName(net)
     static char tempId[100];
 #if defined(EMSCRIPTEN)
  int etext;    
+#elif defined(linux) && defined(__GLIBC__) && defined(__GLIBC_MINOR__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 41)
+    extern char etext;
 #elif defined(linux) && defined(__clang__)
     extern char etext;
 #elif defined(linux) || defined(CYGWIN)

From 2cfe33c47998a996758ebbcd51a509095a2a3b99 Mon Sep 17 00:00:00 2001
From: "Darryl L. Miles" <darryl.miles@darrylmiles.org>
Date: Tue, 15 Jul 2025 23:56:45 +0100
Subject: [PATCH 2/2] netlist.c: whitespace only cleanup

---
 utils/netlist.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/utils/netlist.c b/utils/netlist.c
index b1556c85..9b7d05de 100644
--- a/utils/netlist.c
+++ b/utils/netlist.c
@@ -369,7 +369,7 @@ NLNetName(net)
 {
     static char tempId[100];
 #if defined(EMSCRIPTEN)
- int etext;    
+    int etext;
 #elif defined(linux) && defined(__GLIBC__) && defined(__GLIBC_MINOR__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 41)
     extern char etext;
 #elif defined(linux) && defined(__clang__)
@@ -377,7 +377,7 @@ NLNetName(net)
 #elif defined(linux) || defined(CYGWIN)
     extern int etext asm("etext");
 #elif defined(__APPLE__)
- int etext;
+    int etext;
 #else
     extern int etext;
 #endif
